types/src/hex.rs

```diff
@@ -0,0 +1,74 @@
+//! Hexadecimal serde wrappers for arrays.
+//!
+//! This module provides transparent wrapper types for
+//! arrays that (de)serialize from hexadecimal input/output.
+
+#[cfg(feature = "epee")]
+use cuprate_epee_encoding::{error, macros::bytes, EpeeValue, Marker};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// Wrapper type for a byte array that (de)serializes from/to hexadecimal strings.
+///
+/// # Deserialization
+/// This struct has a custom deserialization that only applies to certain
+/// `N` lengths because [`hex::FromHex`] does not implement for a generic `N`:
+/// <https://docs.rs/hex/0.4.3/src/hex/lib.rs.html#220-230>
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[repr(transparent)]
+pub struct HexBytes<const N: usize>(
+    #[cfg_attr(feature = "serde", serde(with = "hex::serde"))] pub [u8; N],
+);
+
+impl<'de, const N: usize> Deserialize<'de> for HexBytes<N>
+where
+    [u8; N]: hex::FromHex,
+    <[u8; N] as hex::FromHex>::Error: std::fmt::Display,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        Ok(Self(hex::serde::deserialize(deserializer)?))
+    }
+}
```

Comment:
`HexBytes` now uses `<const N: usize>`, deserialization is only blanket implemented if the array supports `hex::serde::deserialization`.